ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -1298,8 +1342,38 @@ generator cd secParam (Model mock hs) =
         , (freqFlush,   pure Flush)
         , (freqRestore, pure Restore)
         , (freqDrop,    Drop <$> QC.choose (0, mockChainLength mock))
+        , (freqGetTables, genGetTables)
         ]
       where
+        genGetTables :: Gen (Cmd ss)
+        genGetTables =
+          let randomPoint = do
+                pt <- QC.arbitrary
+                keys <- ExtLedgerStateTables . TokenToTValue . ApplyKeysMK . DS.Keys <$> QC.arbitrary
+                pure $ GetTablesAtFor pt keys
+          in
+          if null (mockLedger mock)
+            then randomPoint
+            else do
+              QC.frequency
+               [ (5, do
+                     -- existing point, subset of keys
+                     pt <- QC.elements (mockLedger mock)
```

Comment:
This isn't a `pt` is it? Maybe we can pattern match here `(pt, st)` (or sth like that).